test/wallet-rpc-test.js

```diff
@@ -0,0 +1,338 @@
+      assert.equal(response.witness_program, address.hash.toString('hex'));
+    });
+
+    /*
+     * Iterate through the lookahead amount of addresses and request
+     * the rpc 'getaddressinfo' for each, assert that each ismine is
+     * true and then assert that ismine is false when the next address
+     * generated is outside of the lookahead range
+     */
+    it('should detect ismine up to the lookahead', async ($) => {
+      const info = await wclient.getAccount(watchOnlyWalletId, 'default');
+      await wclient.execute('selectwallet', [watchOnlyWalletId]);
+
+      const xpub = testxpub(0, network);
+      for (let i = 0; i < info.lookAhead; i++) {
+        const address = toAddress(xpub, {
+          network,
+          branch: 0,
+          index: i,
+          enc: 'string'
+        });
+
+        const response = await wclient.execute('getaddressinfo', [address]);
+        assert.equal(response.ismine, true);
+      }
+
+      // derive an address at lookahead+1
+      const failed = toAddress(xpub, {
+        network,
+        branch: 0,
+        index: info.lookahead+1,
+        enc: 'string'
+      });
+
+      const response = await wclient.execute('getaddressinfo', [failed]);
+      assert.equal(response.ismine, false);
+    });
+
+    /*
+     * match bitcoind returning an rpc error when sending the
+     * incorrect network
+     */
+    it('should throw for the wrong network', async () => {
+      const xpub = testxpub(0, network);
+      const failed = toAddress(xpub, {
+        network: 'main',
+        branch: 0,
+        index: 0,
+        enc: 'string'
+      });
+
+      const fn = async () => await wclient.execute('getaddressinfo', [failed]);
+      await assert.asyncThrows(fn, 'Invalid address.');
+    });
+
+    it('should fail for invalid address', async () => {
+      const xpub = testxpub(0, network);
+      let failed = toAddress(xpub, {
+        network,
+        branch: 0,
+        index: 0
+      });
+      // remove the first character
+      failed = failed.slice(1, failed.length);
+
+      const fn = async () => await wclient.execute('getaddressinfo', [failed]);
+      await assert.asyncThrows(fn, 'Invalid address.');
+    });
+  });
+});
+
+/*
+ * helper functions
+ */
+function testxpub(index = 0, network) {
+  if (typeof network === 'string')
+    network = Network.get(network);
+  const mnemonic = Mnemonic.fromPhrase(phrase);
+  // m'
+  const priv = HDPrivateKey.fromMnemonic(mnemonic);
+  // m'/44'
+  const bip44Key = priv.derive(44, true);
+  // m'/44'/{type}'
+  const coinType = network.keyPrefix.coinType;
+  const bitcoinKey = bip44Key.derive(coinType, true);
+  // m'/44'/{type}'/${index}'
+  const accountKey = bitcoinKey.derive(index, true);
+  // turn to public key
+  return accountKey.toPublic();
+}
+
+function toAddress(hdpubkey, options) {
+  assert(hdpubkey.depth === 3, 'must pass account level extended key');
+  const {branch,index} = options;
```

Comment:
nit: space after comma is more consistent with codebase for destructuring.

Reply:
👍 good call